paf-mvp-core-js/tests/test-cases/audit-log.test.ts

```diff
@@ -0,0 +1,103 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { buildAuditLog } from '@core/model/audit-log';
+import {
+  AuditLog,
+  IdsAndPreferences,
+  Seed,
+  TransmissionRequest,
+  TransmissionResponse,
+} from '../../src/model/generated-model';
+
+const fixturesDirectory = path.join('tests', 'fixtures', 'audit-log');
+
+interface Fixture {
+  seed: Seed;
+  data: IdsAndPreferences;
+  response: TransmissionResponse;
+  auditLog: AuditLog;
+}
+
+const loadJson = (...filePath: string[]): any => {
+  const joinedPath = path.join(fixturesDirectory, ...filePath);
```

Comment:
Agree with you. I hesitated to do it. Your comment make me consider it seriously.

From my perspective, there is two options:
1. Create ts files instead of json files. I like this one because it catches the issue at compile time and we have reusable models for the fixtures. Far less error-prone to maintain.
2. Keep json files and verify them with schemas. The good side of it is that we have concrete examples that match the documentation.

What do you think ?

Reply:
I'm also in favor of option 1, even if it is very frustrating that we cannot use plain JSON that is more reusable.
But it's a good tradeoff because it's easy to extract JSON from the ts file if we need.